File: 1078-remove-outermost-parentheses/solution.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        flag=False
        x=0
        res=''
        for i in s:
            if not flag and i=='(':
                flag=True
            elif flag and i=='(':
                res=res+'('
                x=x+1
            elif flag and i==')':
                if x>0:
                    x=x-1
                    res=res+')'
                else:
                    flag=False
        return res



                

File: 1078-remove-outermost-parentheses/solution.java
class Solution {
    public String removeOuterParentheses(String s) {
        String res = "";
        boolean flag = false;
        int c = 0;
        for(char i:s.toCharArray()){
            if(i=='('){
                if(!flag){
                    flag = true;
                }
                else{
                    c++;
                    res+="(";
                }
            }else{
                if(c>0){
                    c--;
                    res+=")";
                }else{
                    flag = false;
                }
            }
        }
        return res;
    }
}
